cuda/cuda-test/cu/test.cu

```diff
@@ -13,45 +17,62 @@ void GetCudaProperties() {
   for (int i = 0; i < count; ++i) {
     cudaDeviceProp prop;
     cudaGetDeviceProperties(&prop, i);
+    std::copy(prop.maxGridSize, prop.maxGridSize + 3, maxGridSize);
```

Comment:
investigate and implement how the properties query can be rewritten as an iterator

Reply:
I gave it a try to add an iterator for this but it was rather difficult. I implemented something that resembles an iterator but it might be incorrect.